addons/mail/static/src/core/common/thread_model.js

```diff
@@ -82,8 +82,28 @@ export class Thread extends Record {
         sort: (a1, a2) => (a1.id < a2.id ? 1 : -1),
     });
     activeRtcSession = Record.one("RtcSession");
-    /** @type {object|undefined} */
-    channel;
+    canLeave = Record.attr(false, {
+        /** @this {import("models").Thread} */
+        compute() {
+            return this.computeCanLeave();
+        },
+    });
+    computeCanLeave() {
+        return (
+            ["channel", "group"].includes(this.type) &&
+            !this.message_needaction_counter &&
+            !this.group_based_subscription
+        );
+    }
+    canUnpin = Record.attr(false, {
+        /** @this {import("models").Thread} */
+        compute() {
+            return this.computeCanUnpin();
+        },
+    });
```

Comment:
Every extra field has a non-negligible cost, and especially if it is computed.
I think we might be better off keeping getters for fields we don't actually need to cache because they are very cheap to compute.
And we should aim to have only cheap computes. So we should aim to have no compute.